File: java/src/main/java/com/android/util/threadPool/ThreadPoolManager.java
package com.android.util.threadPool;

/**
 * Created by xuzhb on 2020/4/4
 * Desc:线程池管理器，获取线程池单例
 */
public class ThreadPoolManager {

    private ThreadPoolUtil mFixedThreadPool;
    private ThreadPoolUtil mCachedThreadPool;
    private ThreadPoolUtil mSingleThreadPool;
    private ThreadPoolUtil mScheduledThreadPool;

    private ThreadPoolManager() {
    }

    public static ThreadPoolManager getInstance() {
        return SingleTonHolder.holder;
    }

    private static class SingleTonHolder {
        private static final ThreadPoolManager holder = new ThreadPoolManager();
    }

    public ThreadPoolUtil getFixedThreadPool() {
        if (mFixedThreadPool == null || mFixedThreadPool.isShutDown()) {
            mFixedThreadPool = new ThreadPoolUtil(ThreadPoolUtil.FixedThread, 10);
        }
        return mFixedThreadPool;
    }

    public ThreadPoolUtil getCachedThreadPool() {
        if (mCachedThreadPool == null || mCachedThreadPool.isShutDown()) {
            mCachedThreadPool = new ThreadPoolUtil(ThreadPoolUtil.CachedThread, 0);
        }
        return mCachedThreadPool;
    }

    public ThreadPoolUtil getSingleThreadPool() {
        if (mSingleThreadPool == null || mSingleThreadPool.isShutDown()) {
            mSingleThreadPool = new ThreadPoolUtil(ThreadPoolUtil.SingleThread, 1);
        }
        return mSingleThreadPool;
    }

    public ThreadPoolUtil getScheduledThreadPool() {
        if (mScheduledThreadPool == null || mScheduledThreadPool.isShutDown()) {
            mScheduledThreadPool = new ThreadPoolUtil(ThreadPoolUtil.ScheduledThread, 10);
        }
        return mScheduledThreadPool;
    }

}

File: kotlin/src/main/java/com/android/util/threadPool/ThreadPoolManager.kt
package com.android.util.threadPool

/**
 * Created by xuzhb on 2020/12/8
 * Desc:线程池管理器，获取线程池单例
 */
class ThreadPoolManager private constructor() {

    companion object {
        val instance: ThreadPoolManager = SingleTonHolder.holder
    }

    private object SingleTonHolder {
        val holder = ThreadPoolManager()
    }

    private var mFixedThreadPool: ThreadPoolUtil? = null
    private var mCachedThreadPool: ThreadPoolUtil? = null
    private var mSingleThreadPool: ThreadPoolUtil? = null
    private var mScheduledThreadPool: ThreadPoolUtil? = null

    fun getFixedThreadPool(): ThreadPoolUtil {
        if (mFixedThreadPool == null || mFixedThreadPool!!.isShutDown()) {
            mFixedThreadPool = ThreadPoolUtil(ThreadPoolUtil.FixedThread, 10)
        }
        return mFixedThreadPool!!
    }

    fun getCachedThreadPool(): ThreadPoolUtil {
        if (mCachedThreadPool == null || mCachedThreadPool!!.isShutDown()) {
            mCachedThreadPool = ThreadPoolUtil(ThreadPoolUtil.CachedThread, 0)
        }
        return mCachedThreadPool!!
    }

    fun getSingleThreadPool(): ThreadPoolUtil {
        if (mSingleThreadPool == null || mSingleThreadPool!!.isShutDown()) {
            mSingleThreadPool = ThreadPoolUtil(ThreadPoolUtil.SingleThread, 1)
        }
        return mSingleThreadPool!!
    }

    fun getScheduledThreadPool(): ThreadPoolUtil {
        if (mScheduledThreadPool == null || mScheduledThreadPool!!.isShutDown()) {
            mScheduledThreadPool = ThreadPoolUtil(ThreadPoolUtil.ScheduledThread, 10)
        }
        return mScheduledThreadPool!!
    }

}
